src/main/java/mat/server/service/impl/FhirMeasureServiceImpl.java

```diff
@@ -0,0 +1,113 @@
+package mat.server.service.impl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.support.TransactionTemplate;
+
+import gov.cms.mat.fhir.rest.dto.ConversionOutcome;
+import gov.cms.mat.fhir.rest.dto.ConversionResultDto;
+import mat.client.measure.ManageMeasureDetailModel;
+import mat.client.measure.ManageMeasureSearchModel;
+import mat.client.measure.service.FhirConvertResultResponse;
+import mat.client.measure.service.FhirValidationStatus;
+import mat.client.shared.MatException;
+import mat.client.shared.MatRuntimeException;
+import mat.dao.clause.MeasureDAO;
+import mat.model.clause.Measure;
+import mat.server.service.FhirMeasureService;
+import mat.server.service.MeasureCloningService;
+import mat.server.service.MeasureLibraryService;
+
+@Service
+public class FhirMeasureServiceImpl implements FhirMeasureService {
+
+    private static final Log logger = LogFactory.getLog(FhirMeasureServiceImpl.class);
+
+    private final FhirOrchestrationGatewayService fhirOrchestrationGatewayService;
+
+    private final MeasureLibraryService measureLibraryService;
+
+    private final MeasureCloningService measureCloningService;
+
+    private final MeasureDAO measureDAO;
+
+    private final TransactionTemplate transactionTemplate;
+
+    public FhirMeasureServiceImpl(FhirOrchestrationGatewayService fhirOrchestrationGatewayService, MeasureLibraryService measureLibraryService, MeasureCloningService measureCloningService, MeasureDAO measureDAO, PlatformTransactionManager txManager) {
+        this.fhirOrchestrationGatewayService = fhirOrchestrationGatewayService;
+        this.measureLibraryService = measureLibraryService;
+        this.measureCloningService = measureCloningService;
+        this.measureDAO = measureDAO;
+        this.transactionTemplate = new TransactionTemplate(txManager);
+    }
+
+    @Override
+    public FhirConvertResultResponse convert(ManageMeasureSearchModel.Result sourceMeasure, String loggedinUserId) throws MatException {
```

Comment:
Can we have some documentation for this method. It seems to be the main method responsible for conversion process

Reply:
I think we should prepare a sequence diagram for the conversion